app/front-end/src/features/editor/components/toolbarView/toolbarGroupButtons/applyGroupButtons.tsx

```diff
@@ -1,26 +1,90 @@
+import { ToolbarGroupItem, ToolbarGroupItemProps } from '@/features/editor/components/toolbarView';
+import { useToolbarContext } from '@/features/editor/hooks';
+import { useStatusContext } from '@/hooks';
 import { Deblur as DeblurIcon } from '@mui/icons-material';
+import { useCallback, useMemo } from 'react';
 
-import { ToolbarGroupItemProps } from '@/features/editor/components/toolbarView';
+export interface ApplyGroupButtonsProps {}
 
-const applySpliceAiClick = () => {
-  console.log('Clicked Apply SpliceAI Button!');
-};
+export const ApplyGroupButtons: React.FC<ApplyGroupButtonsProps> = () => {
+  const { blockedStateUpdate } = useStatusContext();
+  const { saveTo, override, applyTo, applyErrorStateUpdate } = useToolbarContext();
 
-const applyCaddClick = () => {
-  console.log('Clicked Apply CADD Button!');
-};
+  const applySpliceAiClick = useCallback(async () => {
+    if (!applyTo) {
+      applyErrorStateUpdate('Please select a file');
```

Comment:
No need for the long text inside a file selector, could just leave the component red

Reply:
Fixed with [MDE/PKFE-31 select changes during file selection](https://github.com/Strexas/kath/pull/57/commits/f758a51ceb9e6c66867e58cfa3697ddbaab254a8)